core/integration/gitcredential_test.go

```diff
@@ -0,0 +1,129 @@
+package core
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/dagger/dagger/testctx"
+	"github.com/stretchr/testify/require"
+)
+
+type GitCredentialSuite struct{}
+
+func TestGitCredential(t *testing.T) {
+	testctx.Run(testCtx, t, GitCredentialSuite{}, Middleware()...)
+}
+
+// TestGitCredentialErrors verifies Git authentication for private modules across different providers
+// using host-specific PATs and isolated Git credentials per test.
+func (GitCredentialSuite) TestGitCredentialErrors(ctx context.Context, t *testctx.T) {
```

Comment:
Could we get one more test case that's similar to the existing ones except that it calls a module and passes a `Directory` as an arg that's from a git repo? e.g. something like:

```
dagger call test --dir https://github.com/grouville/daggerverse-private.git
```

and make sure ops on the git dir work as expected in the module (can list the dir or something like that)?

That's just one tricky case where the resolution with git needs to happen within the CLI+auth-token and after should be usable from modules even if they don't have access to the auth provider directly.